frontend/test/pytest/test_cuda_integration.py

```diff
@@ -0,0 +1,459 @@
+        observed = cuda_compiled()
+        catalyst_compiled = qjit(circuit_catalyst)
+        expected = catalyst_compiled()
+        assert_allclose(expected, observed)
+
+    def test_expval_2(self):
+        """Test multiple_values."""
+        from catalyst.cuda import CudaQDevice
+
+        @qml.qnode(CudaQDevice(wires=2))
+        def circuit():
+            qml.RY(jnp.pi / 4, wires=[1])
+            return qml.expval(qml.PauliZ(1) + qml.PauliX(1))
+
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def circuit_catalyst():
+            qml.RY(jnp.pi / 4, wires=[1])
+            return qml.expval(qml.PauliZ(1) + qml.PauliX(1))
+
+        cuda_compiled = qjit(compiler="cuda_quantum")(circuit)
+        observed = cuda_compiled()
+        catalyst_compiled = qjit(circuit_catalyst)
+        expected = catalyst_compiled()
+        assert_allclose(expected, observed)
+
+    def test_adjoint(self):
+        """Test adjoint."""
+
+        from catalyst.cuda import CudaQDevice
+
+        @qml.qnode(CudaQDevice(wires=2))
+        def circuit():
+            def f():
+                qml.RX(jnp.pi / 23, wires=[0])
+                qml.RX(jnp.pi / 17, wires=[1])
+                qml.Hadamard(wires=[0])
+                qml.Hadamard(wires=[1])
+                qml.PauliX(wires=0)
+                qml.PauliY(wires=1)
+
+            qml.adjoint(f)()
+            return qml.state()
+
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def circuit_catalyst():
+            def f():
+                qml.RX(jnp.pi / 23, wires=[0])
+                qml.RX(jnp.pi / 17, wires=[1])
+                qml.Hadamard(wires=[0])
+                qml.Hadamard(wires=[1])
+                qml.PauliX(wires=0)
+                qml.PauliY(wires=1)
+
+            qml.adjoint(f)()
+            return qml.state()
+
+        cuda_compiled = qjit(compiler="cuda_quantum")(circuit)
+        observed = cuda_compiled()
+        catalyst_compiled = qjit(circuit_catalyst)
+        expected = catalyst_compiled()
+        assert_allclose(expected, observed)
+
+    def test_control_ry(self):
+        """Test control ry."""
+
+        from catalyst.cuda import CudaQDevice
+
+        @qml.qnode(CudaQDevice(wires=2))
+        def circuit():
+            qml.Hadamard(wires=[0])
+            qml.CRY(jnp.pi / 2, wires=[0, 1])
+            return qml.state()
+
+        @qml.qnode(qml.device("lightning.qubit", wires=2))
+        def circuit_catalyst():
+            qml.Hadamard(wires=[0])
+            qml.CRY(jnp.pi / 2, wires=[0, 1])
+            return qml.state()
+
+        cuda_compiled = qjit(compiler="cuda_quantum")(circuit)
+        observed = cuda_compiled()
+        catalyst_compiled = qjit(circuit_catalyst)
+        expected = catalyst_compiled()
+        assert_allclose(expected, observed)
+
+    def test_swap(self):
+        """Test swap."""
+
+        from catalyst.cuda import CudaQDevice
+
+        @qml.qnode(CudaQDevice(wires=2))
```

Comment:
@erick-xanadu is it possible to do

```python
dev = qml.device("cudaq", target="qpp-cpu")
```
or will this require a follow-on PR?

Reply:
But this also might change, I think @dime10 wanted to chat about this with you, which reminds me I need to do an experiment.

Reply:
Thanks!

Reply:
Yep! We can just subclass the main `CudaQDevice` class for each of the above (if/when they are supported)

Reply:
Josh suggested the following which I also like:
```
qml.device("nvidia.custatevec", multi_gpu=True/False)
qml.device("softwareq.qpp")
qml.device("nvidia.cutensornet", mps=True/False)
```

Reply:
⬆️ 👍

Reply:
I added "softwareq.qpp". I do have the others as well but haven't tested them. (I don't think I can test them in my local machine): https://github.com/PennyLaneAI/catalyst/pull/477/commits/3faff901ffa5db85b0d60f5d5743a65acfe1df48 https://github.com/PennyLaneAI/catalyst/pull/477/commits/9ddfe9528ddddd72a258705a2c2cffe4d5230e8a

At the momen't they are unavailable to users until we can test them.